lib/pages/completed_tasks.dart

```diff
@@ -6,90 +6,104 @@ import 'package:task_list_app/constants/constants.dart';
 
-  @override
-  State<CompletedTaskScreen> createState() => _CompletedTaskScreenState();
-}
-
-class _CompletedTaskScreenState extends State<CompletedTaskScreen> {
   @override
   Widget build(BuildContext context) {
-    context.read<TaskCubit>().readTasks();
-    return BlocBuilder<TaskCubit, TaskState>(builder: (context, state) {
-      List<Widget> completedTasks = state.tasks.map(
-        (task) {
-          return ListTile(
-            key: const Key('completed tiles'),
-            leading: IconButton(
-              icon: const Icon(Icons.check),
-              color: Colors.green,
-              onPressed: () {},
-            ),
-            title: Text(
-              task.name,
-              style: tilesText,
-            ),
-            trailing: IconButton(
-              onPressed: () => _onTapTrashIcon(task),
-              icon: const Icon(
-                Icons.delete,
-                color: Colors.redAccent,
+    context.read<TaskCubit>().readTasksByStatus(TaskStatus.completed);
+    return BlocBuilder<TaskCubit, TaskState>(
+      builder: (context, state) {
+        List<Widget> completedTasks = state.tasks.map(
+          (task) {
+            return ListTile(
+              key: const Key('completed_tiles'),
+              leading: const Icon(
+                Icons.check,
+                color: Colors.green,
               ),
-            ),
-          );
-        },
-      ).toList();
+              title: Text(
+                task.name,
+                style: tilesText,
+              ),
+              trailing: IconButton(
+                key: const Key('completed_trash_icon'),
+                onPressed: () => _onTapTrashIcon(context, task),
+                icon: const Icon(
+                  Icons.delete,
+                  color: Colors.redAccent,
+                ),
+              ),
+            );
+          },
+        ).toList();
 
-      return Scaffold(
-        bottomNavigationBar: const BottomNav(),
-        backgroundColor: bgColor,
-        appBar: AppBar(
-          backgroundColor: Colors.black,
-          title: Text(
-            'Completed Tasks',
-            style: TextStyle(color: white),
+        return Scaffold(
+          bottomNavigationBar: const BottomNav(),
+          backgroundColor: bgColor,
+          appBar: AppBar(
+            backgroundColor: Colors.black,
+            title: Text(
+              'Completed Tasks',
+              style: TextStyle(color: white),
+            ),
+            centerTitle: true,
           ),
-          centerTitle: true,
-        ),
-        body: ListView.builder(
-          itemCount: completedTasks.length,
-          itemBuilder: (context, index) {
-            return completedTasks[index];
-          },
-        ),
-      );
-    });
+          body: ListView.builder(
+            itemCount: completedTasks.length,
+            itemBuilder: (context, index) {
+              return completedTasks[index];
+            },
```

Comment:
Consider an arrow function here:

```suggestion
            itemBuilder: (context, index) => completedTasks[index],
```

Reply:
done